src/liballoc/collections/btree/node.rs

```diff
@@ -432,9 +459,9 @@ impl<'a, K, V, Type> NodeRef<marker::Mut<'a>, K, V, Type> {
     /// If the node is an internal node, so not a leaf, it does have all the data a leaf has
     /// (header, keys and values), and this function exposes that.
     ///
-    /// Returns a raw ptr to avoid asserting exclusive access to the entire node.
-    fn as_leaf_mut(&mut self) -> *mut LeafNode<K, V> {
-        self.node.as_ptr()
+    /// We don't need to return a raw ptr because we have exclusive access to the entire node.
```

Comment:
This is bold, I would not have dared.^^ But if Miri doesn't complain I guess it is okay.

Reply:
Well, I believe it was Mark who reminded me in some review that `marker::Mut` stuff was supposed to have exclusive references. But now I realize it's a contradictory "Surely I have exclusive access. Here's a second reference; surely you know that you definitely can't call anyone else as arrogant as me".

Reply:
This method consumes the handle though, right? So the returned reference is still unique?

Reply:
Err whaat? `fn as_leaf_mut(&mut self)` consumes?

You mean that it should consume? It does seem like a lot of places could do that. Or they just need to adjust node length without accessing the whole leaf.

Reply:
Sorry, it doesn't consume, but it reborrows. As in the lifetime of the returned reference equals the lifetime of the argument reference, much like e.g. `Cell::get_mut`.

Reply:
Right, and that's what is explicitly explained in `keys_mut` and `vals_mut`. Also, I guess "unique access" is more Rusticious than the "exclusive access" I put in comments.